Jenkinsfile-asf

```diff
@@ -0,0 +1,84 @@
+#!groovy
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+def executeTests() {
+    def testJavaHome = sh(label: 'Get TEST_JAVA_HOME', script: "jabba which ${TEST_JAVA_VERSION}", returnStdout: true).trim()
+    def testJavaVersion = (TEST_JAVA_VERSION =~ /.*@1\.(\d+)/)[0][1]
+    def serverVersion = env.SERVER_VERSION
+    sh '''
+    	. ~/env.txt
+   	printenv | sort
+  	mvn -B -V install -DskipTests -Dmaven.javadoc.skip=true
+   	jabba use ${TEST_JAVA_VERSION}
+    	printenv | sort
+	mvn -B -V verify -T 1 -Dcassandra.version=''' + serverVersion + ''' -Ddse=false -Dmaven.test.failure.ignore=true -Dmaven.javadoc.skip=true'''
+}
+
+// branch pattern for cron
+// should match 3.x, 4.x, 4.5.x, etc
+def branchPatternCron() {
+  ~"((\\d+(\\.[\\dx]+)+))"
+}
+
+pipeline {
```

Comment:
nit: my preference and request is we move the `pipeline { … }` section to the beginning of the file (after the license).

as a reader, i find it easier to read (i.e. more intuitive) to start with the overview the pipeline provides, and then afterwards dive into each supporting function that exists.  this approach is taken in https://github.com/apache/cassandra/blob/trunk/.jenkins/Jenkinsfile

Reply:
can we move `executeTests()` below please (like it is in https://github.com/apache/cassandra-java-driver/pull/1956 )